### PR TITLE
Move get_sha to function

### PR DESCRIPTION
**Pull request by AdamBrousseau**
- Also pass branch to ls-remote command
- This is better than grep because it will only
  return the branch you specified
- Prevents multiple SHAs form being returned

Signed-off-by: Adam Brousseau <adam.brousseau88@gmail.com>

### CODE REVIEW

**Comment by AdamBrousseau** [comment]
Example of failure (returns multiple SHAs)
https://ci.eclipse.org/openj9/job/Pipeline-Build-Test-JDK8-linux_390-64_cmprssptrs/85/console
Build description:
```
OpenJ9: d2bf210
OMR: b9d1250 1f7d88e
OpenJDK: 2efea45
```
Console:
```
[omr] Running shell script
+ git ls-remote https://github.com/eclipse/openj9-omr.git
+ grep refs/heads/openj9
+ awk '{ print $1 }'
+ sed -r 's/(^.{7}).*/\1/'
[Pipeline] }
[Pipeline] // dir
[Pipeline] echo
OPENJDK_SHA:2efea45
[Pipeline] echo
OPENJ9_SHA:d2bf210
[Pipeline] echo
OMR_SHA:b9d1250
1f7d88e
```

Which causes get_source to fail
https://ci.eclipse.org/openj9/job/Build-JDK8-linux_390-64_cmprssptrs/87/console
```
00:00:33.633 Update omr to commit ID: b9d1250
00:00:33.633 
00:00:33.633 Note: checking out 'b9d1250'.
00:00:33.633 
00:00:33.633 You are in 'detached HEAD' state. You can look around, make experimental
00:00:33.633 changes and commit them, and you can discard any commits you make in this
00:00:33.633 state without impacting any branches by performing another checkout.
00:00:33.633 
00:00:33.633 If you want to create a new branch to retain commits you create, you may
00:00:33.633 do so (now or later) by using -b with the checkout command again. Example:
00:00:33.633 
00:00:33.633   git checkout -b <new-branch-name>
00:00:33.633 
00:00:33.633 HEAD is now at b9d1250... Merge pull request #2268 from Varun-garg/fixSignalTest
00:00:33.633 /home/jenkins/jenkins-agent/workspace/Build-JDK8-linux_390-64_cmprssptrs
00:00:33.633 
00:00:33.633 Update openj9 to commit ID: d2bf210
00:00:33.633 
00:00:33.929 Note: checking out 'd2bf210'.
00:00:33.930 
00:00:33.930 You are in 'detached HEAD' state. You can look around, make experimental
00:00:33.930 changes and commit them, and you can discard any commits you make in this
00:00:33.930 state without impacting any branches by performing another checkout.
00:00:33.930 
00:00:33.930 If you want to create a new branch to retain commits you create, you may
00:00:33.930 do so (now or later) by using -b with the checkout command again. Example:
00:00:33.930 
00:00:33.930   git checkout -b <new-branch-name>
00:00:33.930 
00:00:33.930 HEAD is now at d2bf210... Merge pull request #1168 from eric98zhang/lworldrefactor
00:00:33.930 /home/jenkins/jenkins-agent/workspace/Build-JDK8-linux_390-64_cmprssptrs
00:00:33.930 + 1f7d88e
00:00:33.930 /home/jenkins/jenkins-agent/workspace/Build-JDK8-linux_390-64_cmprssptrs@tmp/durable-9424af65/script.sh: 3: /home/jenkins/jenkins-agent/workspace/Build-JDK8-linux_390-64_cmprssptrs@tmp/durable-9424af65/script.sh: 1f7d88e: not found
```

**Comment by AdamBrousseau** [comment]
Tested out of my fork in this build
https://ci.eclipse.org/openj9/job/Pipeline-Build-Test-JDK8-linux_390-64_cmprssptrs/86/consolehttps://ci.eclipse.org/openj9/job/Pipeline-Build-Test-JDK8-linux_390-64_cmprssptrs/86/
```
[omr] Running shell script
+ git ls-remote https://github.com/eclipse/openj9-omr.git refs/heads/openj9
+ awk '{ print $1 }'
+ sed -r 's/(^.{7}).*/\1/'
[Pipeline] }
[Pipeline] // dir
[Pipeline] echo
OPENJDK_SHA:4a52792
[Pipeline] echo
OPENJ9_SHA:9c6d60f
[Pipeline] echo
OMR_SHA:b9d1250
```

File: buildenv/jenkins/pipeline-functions [hunk 1]
```diff
@@ -51,6 +40,13 @@
     }
 }
 
+def get_sha(REPO, BRANCH) {
```
Comment (DanHeidinga) [inline buildenv/jenkins/pipeline-functions]: Can you comment this to describe what it does and why it passes both repo & branch?

Comment (AdamBrousseau) [reply buildenv/jenkins/pipeline-functions]: Pushed new commit with some comments

File: buildenv/jenkins/pipeline-functions [hunk 2]
```diff
@@ -51,6 +42,16 @@
...
+    // Get the SHA at the tip of the BRANCH in REPO.
+    // Allows Pipelines to kick off multiple builds and have the same SHA built everywhere.
+    return sh (
+            // "git ls-remote $REPO" will return all refs, adding "$BRANCH" will only return the specific branch we are interested in
+            script: "git ls-remote $REPO refs/heads/$BRANCH | awk \'{ print \$1 }\' | sed -r 's/(^.{7}).*/\\1/'",
```
Comment (keithc-ca) [inline buildenv/jenkins/pipeline-functions]: Why the change to only return the first 7 characters of the SHA?

The `sed` filter eliminates the value of the `awk` part of the pipeline.

Comment (AdamBrousseau) [reply buildenv/jenkins/pipeline-functions]: The only thing I changed in the commands was removing the grep and adding the branch name to the git command. I agree we should just use the cut and eliminate the awk, will update.

**Comment by AdamBrousseau** [comment]
Fixed

**Comment by AdamBrousseau** [comment]
Must have missed adding it to the index, fixed now

**Comment by DanHeidinga** [comment]
When I was validating the command on one my repos, I noticed `git log --oneline` shows 8 character shas.  @AdamBrousseau Do you know why we're picking 7?

**Comment by AdamBrousseau** [comment]
Not much difference between 7 and 8 chars in a SHA.
https://git-scm.com/book/en/v2/Git-Tools-Revision-Selection#Short-SHA-1
>Generally, eight to ten characters are more than enough to be unique within a project. For example, as of October 2017, the Linux kernel (which is a fairly sizable project) has over 700,000 commits and almost six million objects, with no two objects whose SHA-1s are identical in the first 11 characters.

I think `--short` used to give 7 chars but locally I am now on git 2.16.1 and it gives me 10. I can change it to 8 or whatever we decide but 7 is likely good enough for a while yet.

**Comment by keithc-ca** [comment]
I had hoped that `ls-remote` had an option to automatically abbreviate SHAs (but not so much as to make them ambiguous), but I couldn't find it. If someone else knows of such an option, I think we should use it.

**Comment by DanHeidinga** [comment]
Not running a PR build as this change will only be tested by the nightly build
